test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Caching/CachingTests.cs

```diff
@@ -232,10 +242,55 @@ public class CachingTests
                 server);
 
             // Assert
-            validations.Assert(CachingValidationType.CommandSucceeded, true);
-            validations.Assert(CachingValidationType.PackageInstalled, true);
-            validations.Assert(CachingValidationType.PackageFromGlobalPackagesFolderUsed, success);
-            validations.Assert(CachingValidationType.PackageFromSourceNotUsed, success);
+            foreach (var validation in validations)
+            {
+                validation.Assert(CachingValidationType.CommandSucceeded, true);
+                validation.Assert(CachingValidationType.PackageInstalled, true);
+                validation.Assert(CachingValidationType.PackageFromGlobalPackagesFolderUsed, success);
+                validation.Assert(CachingValidationType.PackageFromSourceNotUsed, success);
+            }
+        }
+
+        [Theory]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.Default, ServerType.V2, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.Default, ServerType.V3, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache, ServerType.V2, true, true)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache, ServerType.V3, true, true)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.DirectDownload, ServerType.V2, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.DirectDownload, ServerType.V3, true, false)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache | CachingType.DirectDownload, ServerType.V2, true, true)]
+        [InlineData(typeof(RestoreProjectJsonCommand), CachingType.NoCache | CachingType.DirectDownload, ServerType.V3, true, true)]
+        public async Task NuGetExe_Caching_DoesNotNoOp(Type type, CachingType caching, ServerType server, bool success, bool doesNotNoOp)
```

Comment:
Can we invert the variable name and invert the booleans? `noOps` seems a bit more terse.